### PR TITLE
Prevent Python 3.7 from being unsupported

### PR DESCRIPTION
**Pull request by sigmavirus24**
Python 3.7.0 is supported but not preferred given how new it is. As a
result, we don't want it to be the default, but we also don't want users
to be confused when upgrading to it.

Closes gh-728

### CODE REVIEW

File: bin/steps/python [hunk 1]
```diff
@@ -14,12 +14,17 @@ if [[ $PYTHON_VERSION =~ ^python-2 ]]; then
     echo "       Learn More: https://devcenter.heroku.com/articles/python-runtimes"
   fi
 else
-  if [[ "$PYTHON_VERSION" != "$LATEST_3" ]]; then
-    puts-warn "The latest version of Python 3 is $LATEST_3 (you are using $PYTHON_VERSION, which is unsupported)."
-    puts-warn "We recommend upgrading by specifying the latest version ($LATEST_3)."
+  if [[ $PYTHON_VERSION =~ ^python-3.7 ]] && [[ "$PYTHON_VERSION" != "$LATEST_37" ]]; then
+    puts-warn "The latest version of Python 3 is $LATEST_37 (you are using $PYTHON_VERSION, which is unsupported)."
```
Comment (edmorley) [inline bin/steps/python]: Should this say "of Python 3.7" (and similarly for 3.6 below)? :-)

**Comment by DEKHTIARJonathan** [comment]
@sigmavirus24 @KevinBrolly I keep getting the same error.

See : https://github.com/heroku/heroku-buildpack-python/issues/728#issuecomment-403751189

**Comment by KevinBrolly** [comment]
@DEKHTIARJonathan You are actually seeing two separate problems, the first one is solved by this PR:

```
!     The latest version of Python 3 is python-3.6.6 (you are using python-3.7.0, which is unsupported).
!     We recommend upgrading by specifying the latest version (python-3.6.6).
      Learn More: https://devcenter.heroku.com/articles/python-runtimes
```

When this PR is released the message will correctly indicate that you are on the latest version.

The second problem is this:

```
 !     Requested runtime (python-3.7.0) is not available for this stack (cedar-14).
```

That however is a bit of a head-scratcher as python-3.7.0 is available for cedar-14, I have checked this myself and you should not see this error.  I noticed at the bottom of your error message the following:

`Failed to compile droplet: Failed to compile droplet: exit status 1`

Are you running the buildpack on DigitalOcean? The only thing I can think of that could be the problem is that DigitalOcean may be using a different `BUILDPACK_VENDOR_URL` on their platform that does not have python-3.7.0, could you perhaps open a support request with them and check?

**Comment by DEKHTIARJonathan** [comment]
I'm actually using +/- raw CF on IBM Bluemix and I use Python 3.6.6 on production.

```yaml
defaults: &defaults
  domain: eu-gb.mybluemix.net
  timeout: 120
  memory: 768M
  disk_quota: 2G
  path: .
  stack: cflinuxfs2
  # buildpack: https://github.com/heroku/heroku-buildpack-python.git  # does not work
  buildpack: https://github.com/cloudfoundry/buildpack-python.git     # works perfectly
```

runtime.txt
```
python-3.6.6
```

**Comment by sigmavirus24** [comment]
So it sounds like you're having issues using a different stack image (which likely sets a different STACK variable) with our buildpack and want us to be completely compatible with a moving target (the cloudfoundry image/buildpack). Am I missing something here?

**Comment by DEKHTIARJonathan** [comment]
Oh maybe I missed something, I thought you were using the same stack.
Which stack shall I use with this BP ?

**Comment by sigmavirus24** [comment]
You can use any of the three that we test against: https://travis-ci.org/heroku/heroku-buildpack-python/builds/402204669, e.g., `heroku/heroku:18-build`. The full list is here: https://hub.docker.com/r/heroku/heroku/

**Comment by DEKHTIARJonathan** [comment]
Oh my bad. I try this immediately. Thanks for the help

**Pull request merged**